packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -166,8 +166,58 @@ class _RunMetrics {
   }
 }
 
+
+/// How the child is inscribed into the available space.
+///
+/// See also:
+///
+///  * [Wrap], the wrap widget.
+///  * [Wrapped], the widget to set the [WrapFit].
+enum WrapFit {
+  /// Determines whether to place the child in the current or the
+  /// next run, depending on its minimum intrinsic size.
+  ///
+  /// The child is stretched to fill the entire run
+  /// (this does not apply if the [Wrap] is unconstrained).
+  ///
+  /// This is the default fit of the [Wrapped] widget.
+  ///
+  /// This setting can be expensive (see [RenderBox.getMinIntrinsicWidth]).
+  fillRun(true),
+
+  /// Determines whether to place the child in the current or the
+  /// next run, depending on its minimum intrinsic size.
+  ///
+  /// The child is stretched to fill the entire run if the next child starts a
+  /// new run (this does not apply if the [Wrap] is unconstrained).
+  ///
+  /// This setting can be expensive (see [RenderBox.getMinIntrinsicWidth]).
+  mightFill(false),
+
+  /// Determines whether to place the child in the current or the
+  /// next run, depending on its minimum intrinsic size.
+  ///
+  /// This setting can be expensive (see [RenderBox.getMinIntrinsicWidth]).
+  constrained(false),
+
+  /// The child is placed in a new run and stretched to fill the
+  /// available space.
+  ///
+  /// (This does not apply if the [Wrap] is unconstrained.)
+  fillNext(true);
+
+  const WrapFit(this.isTight);
+
+  /// `true` if the [WrapFit] always forces the child to fill the assigned run.
+  final bool isTight;
+}
+
```

Comment:
Another way this could have been approached is to model this after `Flexible` [1]: We assign children to runs as usual and if we encounter a flexible child we ask it for its min intrinsics. If it still fits in the current run, we place it there, if not, it starts a new run. One a run is full of children, we have to determine the size of the flexible ones by taking the space not used up by non-flexible children and dividing it up among the flexible ones according to their flex factors.

This would allow for use cases where the remaining space should be used up by children that aren't last in their run. It doesn't allow for "filling the run only if it is the last child", but I am not sure on how useful that particular setting actually is.

The underlying question here is: What use cases are we trying to cover here?

[1] https://main-api.flutter.dev/flutter/widgets/Flexible-class.html

Reply:
I concur with this feedback—the example shown in https://pub.dev/packages/flexible_wrap would be nice to have as part of this PR.